lightning/src/ln/features.rs

```diff
@@ -442,6 +478,21 @@ impl InvoiceFeatures {
 	}
 }
 
+impl ChannelTypeFeatures {
+	/// Constructs the implicit channel type based on the common supported types between us and our
+	/// counterparty
+	pub(crate) fn from_counterparty_init(counterparty_init: &InitFeatures) -> Self {
+		counterparty_init.to_context_internal()
```

Comment:
If they only *support* static_remotekey (i.e., they don't require it), should we translate their features to require static_remotekey?

Reply:
The BOLT says every bit uses the even/required form, so I dunno if we should? We do consider a channel type features object as "dont know it" if it has odd bits, I believe.

https://github.com/lightningnetwork/lightning-rfc/commit/4372f9d5380005a0e860875cf403cc0ee4211759#diff-ed04ca2c673fd6aabde69389511fa9ee60cb44d6b2ef6c88b549ffaa753d6afeR206-R214